### PR TITLE
TST: Add pdf_reader_page fixture

### PR DESCRIPTION
**Pull request by MartinThoma**
This change should make our tests shorter and easier to read.

### CODE REVIEW

**Comment by MartinThoma** [comment]
@pubpub-zz What to you think about this type of change? I've only applied this to one of up to 52 places where it might be used. I didn't do the rest as I want to know your opinion before I continue.

We pretty often need just a random page for our tests. This could make it more obvious that there is nothing special about that page.

If anybody reads the tests to understand how a certain feature is used, fixtures might make their lives harder. For us, it might make it more clear what we actually want to test.

**Review by pubpub-zz — APPROVED** [review]
I like your proposal with just one comment (inline)

Just one point : do you have any idea to improve readability of all the test files using urls?

File: tests/conftest.py [hunk 1]
```diff
@@ -15,3 +22,12 @@ def pdf_file_path(tmp_path_factory):
 def txt_file_path(tmp_path_factory):
     fn = tmp_path_factory.mktemp("pypdf-data") / f"{uuid.uuid4()}.txt"
     return fn
+
+
+@pytest.fixture(scope="session")
+def pdf_reader_page():
```
[the remark this answers is not in the document]
Comment (MartinThoma) [reply tests/conftest.py:28]: Yes, I was wondering about this as well. I thought that this fixture should only be used in places where the concrete page / PDF could be swapped without affecting the test. Hence I would like to keep the name.

Another idea was to parametrize the fixture to allow specifying a specific (local) PDF / page. But that might actually harm readability.

**Comment by MartinThoma** [comment]
> do you have any idea to improve readability of all the test files using urls?

Instead of 

```python
@pytest.mark.enable_socket()
@pytest.mark.slow()
def test_compute_space_width():
    url = "https://corpora.tika.apache.org/base/docs/govdocs1/923/923406.pdf"
    name = "tika-923406.pdf"

    reader = PdfReader(BytesIO(get_pdf_from_url(url, name=name)))
    for page in reader.pages:
        page.extract_text()
```

we could do 

```python
@pytest.mark.enable_socket()
@pytest.mark.slow()
@pytest.mark.parametrize(
    "external_url_reader",
    [
        [
            "https://corpora.tika.apache.org/base/docs/govdocs1/923/923406.pdf",
            "tika-923406.pdf",
        ]
    ],
    indirect=True,
)
def test_compute_space_width(external_url_reader):
    reader = external_url_reader
    for page in reader.pages:
        page.extract_text()
```

I don't think that is better, though :sweat_smile:

**Comment by MartinThoma** [comment]
> do you have any idea to improve readability of all the test files using urls?

I was hoping that we could get more PDFs into https://github.com/py-pdf/sample-files so that we don't need the load anything from URLs anymore.

**Comment by pubpub-zz** [comment]
> > do you have any idea to improve readability of all the test files using urls?
> 
> I was hoping that we could get more PDFs into https://github.com/py-pdf/sample-files so that we don't need the load anything from URLs anymore.

I like to use the pdf which has produced the error, it is so much easier to ensure the test coverage is good

**Comment by MartinThoma** [comment]
Added via https://github.com/py-pdf/pypdf/pull/1738

**Pull request closed without merging**